app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,13 @@
+class SessionsController < ApplicationController   
+  def create
+    auth = request.env["omniauth.auth"]     
+    user = User.find_by_provider_and_uid(auth["provider"], auth["uid"]) || User.create_with_omniauth(auth)
+    session[:user_id] = user.id
+    redirect_to root_url, :notice => "Signed in!"
+  end
+
+  def destroy
+    session[:user_id] = nil
+    redirect_to root_url, :notice => "Signed out!"
```

Comment:
I think we still have to redirect; just to the UI url. Because the request isn't made by the client; it's made by GH.